### PR TITLE
Standardize dataset directory structure (Issue #209)

### PR DESCRIPTION
**Pull request by ng390**
Closes Issue #209. Note that users who already have downloaded external datasets may have additional downloads given the new paths.

### CODE REVIEW

**Comment by davidslater** [comment]
These changes look good. A couple of questions.

I have a couple of other folders in my datasets directory: "downloads" and "pytorch". Do you have these? And are they primarily for downloading/saving model weights, or are they dataset related?

**Comment by ng390** [comment]
> These changes look good. A couple of questions.
> 
> I have a couple of other folders in my datasets directory: "downloads" and "pytorch". Do you have these? And are they primarily for downloading/saving model weights, or are they dataset related?

I don't currently have a pytorch directory, but it looks like it is cached model weights based on the source code. The downloads directory looks like it stores cached information from the in-memory datasets CIFAR and MNIST.

**Comment by ng390** [comment]
As part of this issue should move the downloads directory into the appropriate subdirectories. The cached weights may be part of Issue #171?

**Comment by davidslater** [comment]
Okay, then it should be solved with a different PR.

Can you add some material to the datasets.py docstring stating that:
1) each standard dataset resides in its own subdir under `dataset_dir` based on the name of the function in the datsets file
2) the `download` subdir is reserved for caching
3) the `private` subdir is reserved for private datasets

**Comment by davidslater** [comment]
For `download`, we may not have that much control over where it goes - I think it's a TensorFlow thing - so I would probably leave it where it's at.

File: armory/data/datasets.py [hunk 1]
```diff
@@ -1,5 +1,9 @@
 """
 Download and preprocess common datasets.
+Each standard dataset resides in its own subdirectory under dataset_dir based
+upon the name of the function in the datasets file.
+The download subdirectory under datset_dir is reserved for caching.
```
Comment (davidslater) [inline armory/data/datasets.py]: nit: datset_dir is misspelled. 

Also, add quotes or ticks around the specific subdirectory names. 

Add an example to the standard dataset, (e.g., "cifar10")

File: armory/data/datasets.py [hunk 2]
```diff
@@ -1,5 +1,10 @@
 """
 Download and preprocess common datasets.
+Each standard dataset resides in its own subdirectory under dataset_dir based
+upon the name of the function in the datasets file. For example, the cifar10
+data is found at 'dataset_dir/cifar10'
+The 'download' subdirectory under dataset_dir is reserved for caching.
+The 'private' subdirectory under dataset_dir is reserved for private datasets.
```
Comment (davidslater) [inline armory/data/datasets.py]: Sorry, one more! (I think that this is the last one.) "dataset_dir" is not an actual directory name, but a placeholder for a key in the armory config. I would replace every instance of dataset_dir with <dataset_dir> to make it clear that it's not a string literal.

**Pull request merged**